homeassistant/components/velbus/services.py

```diff
@@ -92,41 +117,100 @@ async def clear_cache(call: ServiceCall) -> None:
         DOMAIN,
         SERVICE_SCAN,
         scan,
-        vol.Schema({vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id)}),
+        vol.Any(
+            vol.Schema(
+                {
+                    vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id),
+                }
+            ),
+            vol.Schema(
+                {
+                    vol.Required(CONF_CONFIG_ENTRY): selector.ConfigEntrySelector(
+                        {
+                            "integration": DOMAIN,
+                        }
+                    )
+                }
+            ),
+        ),
     )
 
     hass.services.async_register(
         DOMAIN,
         SERVICE_SYNC,
         syn_clock,
-        vol.Schema({vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id)}),
+        vol.Any(
+            vol.Schema(
+                {
+                    vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id),
+                }
+            ),
+            vol.Schema(
+                {
+                    vol.Required(CONF_CONFIG_ENTRY): selector.ConfigEntrySelector(
+                        {
+                            "integration": DOMAIN,
+                        }
+                    )
+                }
+            ),
+        ),
     )
 
     hass.services.async_register(
         DOMAIN,
         SERVICE_SET_MEMO_TEXT,
         set_memo_text,
-        vol.Schema(
-            {
-                vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id),
-                vol.Required(CONF_ADDRESS): vol.All(
-                    vol.Coerce(int), vol.Range(min=0, max=255)
-                ),
-                vol.Optional(CONF_MEMO_TEXT, default=""): cv.template,
-            }
+        vol.Any(
+            vol.Schema(
+                {
+                    vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id),
+                    vol.Required(CONF_ADDRESS): vol.All(
+                        vol.Coerce(int), vol.Range(min=0, max=255)
+                    ),
+                    vol.Optional(CONF_MEMO_TEXT, default=""): cv.template,
+                }
+            ),
+            vol.Schema(
+                {
+                    vol.Required(CONF_CONFIG_ENTRY): selector.ConfigEntrySelector(
+                        {
+                            "integration": DOMAIN,
+                        }
+                    ),
+                    vol.Required(CONF_ADDRESS): vol.All(
+                        vol.Coerce(int), vol.Range(min=0, max=255)
+                    ),
+                    vol.Optional(CONF_MEMO_TEXT, default=""): cv.template,
+                }
+            ),
         ),
```

Comment:
I believe you can wrap this in a `vol.All` with all the overlapping keys, and then have the 2 others in the `vol.Any` inside of it

Reply:
after some discussion on discord we decided to keep it as is.